src/picongpu/include/particles/bremsstrahlung/Bremsstrahlung.tpp

```diff
@@ -0,0 +1,252 @@
+              cachedIonCounter,
+              fieldIonCounterBlock
+              );
+
+    /* wait for shared memory to be initialized */
+    __syncthreads();
+
+    /* initialize random number generator with the local cell index in the simulation */
+    this->randomGen.init(localCellOffset);
+}
+
+
+template<typename T_IonSpecies, typename T_ElectronSpecies, typename T_PhotonSpecies>
+DINLINE
+float3_X Bremsstrahlung<T_IonSpecies, T_ElectronSpecies, T_PhotonSpecies>::scatterByTheta
+    (const float3_X vec, const float_X theta)
+{
+    using namespace PMacc::algorithms;
+
+    float_X sinTheta, cosTheta;
+    math::sincos(theta, sinTheta, cosTheta);
+
+    const float_X phi = -float_X(M_PI) + float_X(2.0) * float_X(M_PI) * this->randomGen();
+    float_X sinPhi, cosPhi;
+    math::sincos(phi, sinPhi, cosPhi);
+
+    const float3_X vecUp(0.0, 0.0, 1.0);
+    float3_X vecOrtho1 = math::cross(vecUp, vec);
+    const float_X vecOrtho1Abs = math::abs(vecOrtho1);
+
+    float3_X vecOrtho1_norm;
+    if(vecOrtho1Abs == float_X(0.0))
+        vecOrtho1_norm = float3_X(1.0, 0.0, 0.0);
+    else
+        vecOrtho1_norm = vecOrtho1 / vecOrtho1Abs;
+    const float3_X vecOrtho2 = math::cross(vecOrtho1_norm, vec);
+    vecOrtho1 = vecOrtho1_norm * math::abs(vec);
+
+    return vec * cosTheta +
+           vecOrtho1 * (sinTheta * cosPhi) +
+           vecOrtho2 * (sinTheta * sinPhi);
+}
+
+template<typename T_IonSpecies, typename T_ElectronSpecies, typename T_PhotonSpecies>
+DINLINE
+unsigned int Bremsstrahlung<T_IonSpecies, T_ElectronSpecies, T_PhotonSpecies>::numNewParticles(FrameType& sourceFrame, int localIdx)
+{
+    using namespace PMacc::algorithms;
+
+    PMACC_AUTO(particle, sourceFrame[localIdx]);
+
+    /* particle position, used for field-to-particle interpolation */
+    const floatD_X pos = particle[position_];
+    const int particleCellIdx = particle[localCellIdx_];
+    /* multi-dim coordinate of the local cell inside the super cell */
+    DataSpace<TVec::dim> localCell(DataSpaceOperations<TVec::dim>::template map<TVec > (particleCellIdx));
+    /* interpolation of fieldTmp */
+    const fieldSolver::numericalCellType::traits::FieldPosition<FieldTmp, simDim> fieldTmpPos;
+    const ValueTypeIonCounter ionCount_norm = Field2ParticleInterpolation()
+        (cachedIonCounter.shift(localCell).toCursor(), pos, fieldTmpPos());
+
+    float_X ionDensity = ionCount_norm.x() *
+        (static_cast<float_X>(particleToGrid::derivedAttributes::Counter().getUnit().x()) / CELL_VOLUME);
+
+    const float_X weighting = particle[weighting_];
+
+    const float_X c = SPEED_OF_LIGHT;
+    float3_X mom = particle[momentum_] / weighting;
+    const float_X momAbs = math::abs(mom);
+    float3_X mom_norm = mom / momAbs;
+
+    const float_X mass = frame::getMass<FrameType>();
+    const float_X momAbs_norm = momAbs / (mass * c);
+    const float_X Ekin = (math::sqrt(float_X(1.0) + momAbs_norm*momAbs_norm) - float_X(1.0)) * mass * c*c;
```

Comment:
I know, I just want to avoid code duplication and es mentioned before automatically include improvements done in the general gamma calculation.